state/state.go

```diff
@@ -44,9 +41,7 @@ func calcABCIResponsesKey(height int64) []byte {
 // but the fields should only be changed by calling state.SetBlockAndValidators.
 // NOTE: not goroutine-safe.
```

Comment:
If we're going to have a convention of allowing non-pointer State, (which is nice), then we should also change our State methods to take non-pointer receivers.  The downside is that we can't memoize anything on State anymore, but it looks like we're not doing that anyways.

It's also potentially slower, but I don't think it's enough to worry about, State is overall less than 200 bytes...